lib/authentication/authentication.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:admin_dashboard/authentication/authentication_service.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:github_sign_in/github_sign_in.dart';
+
+class FirebaseAuthenticationService implements AuthenticationService{
+  final FirebaseAuth _firebaseAuth = FirebaseAuth.instance;
+
+  @override
+  Future<String?> emailLogin(String email, String password) async{
+    UserCredential userCredential = await _firebaseAuth.signInWithEmailAndPassword(
+        email: email, password: password
+    );
+    return userCredential.user?.uid;
+  }
+
+  @override
+  Future<String?> githubLogin(BuildContext context) async {
+    // Create a GitHubSignIn instance
+    final GitHubSignIn gitHubSignIn = GitHubSignIn(
+        clientId: "a0a55c81922cc8cb2e14",
+        clientSecret: "7a58f75f116100000550f299915feb672c342ce9",
```

Comment:
Where did you get this? and why this is in the comment? this is secret and must be passed via environment perhaps, isn't it?